File: software/examples/php/ExampleCallback.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletRotaryPotiV2.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletRotaryPotiV2;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Rotary Poti Bricklet 2.0

// Callback function for position callback
function cb_position($position)
{
    echo "Position: $position °\n";
}

$ipcon = new IPConnection(); // Create IP connection
$rp = new BrickletRotaryPotiV2(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register position callback to function cb_position
$rp->registerCallback(BrickletRotaryPotiV2::CALLBACK_POSITION, 'cb_position');

// Set period for position callback to 0.25s (250ms) without a threshold
$rp->setPositionCallbackConfiguration(250, FALSE, 'x', 0, 0);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Rotary Poti Bricklet 2.0

	// Callback function for position callback
	static void PositionCB(BrickletRotaryPotiV2 sender, short position)
	{
		Console.WriteLine("Position: " + position + " °");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRotaryPotiV2 rp = new BrickletRotaryPotiV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register position callback to function PositionCB
		rp.PositionCallback += PositionCB;

		// Set period for position callback to 0.25s (250ms) without a threshold
		rp.SetPositionCallbackConfiguration(250, false, 'x', 0, 0);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
